starter-code/app.js

```diff
@@ -1,5 +1,76 @@
+/*jshint esversion: 6*/
+
 const express = require('express');
 const app = express();
 const Chuck  = require('chucknorris-io');
 const client = new Chuck();
+const bodyParser = require('body-parser');
+
+
+app.set('views', __dirname + '/views');
+app.set('view engine','ejs');
+app.use(bodyParser.urlencoded({ extended: true }));
+
+
+app.get('/',(req , res, next)=>{
+  res.render('index');
+});
+
+
+app.get('/random', (req, res, next)=> {
+
+  // Retrieve a random chuck joke
+  client.getRandomJoke()
+    .then((response) => {
+      // let temp = response.value;
+      res.render('random',{
+        joke: response.value
+      });
+      // console.log(response.value);
+    }).catch((err) => {
+      res.send('Chuck is not here right now - please leave a message');
+    });
+});
+
+app.get('/categories', (req, res, next)=> {
+
+  client.getJokeCategories()
```

Comment:
why you don't just write instead of:
.then((response)=>  {
 +      let array = response;
 +      res.render('categories',{categories: array});

just:
.then((categories)=>  {
 res.render('categories', {categories});